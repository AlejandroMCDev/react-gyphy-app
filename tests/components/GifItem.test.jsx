import { render,screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem";



describe('GifItem prueba', () => {
    const title = "Elden Ring";
    const url = "https://eldenring.com/";

    test('Debe hacer match con el snapshot ', () => {
        const {container} = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar la imagen con el URL y alt indicado', () => {
        render(<GifItem title={title} url={url} />);

        const {src,alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
    
    
    
})
