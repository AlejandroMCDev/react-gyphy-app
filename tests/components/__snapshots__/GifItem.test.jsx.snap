// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem prueba Debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Elden Ring"
      src="https://eldenring.com/"
    />
    <p>
      Elden Ring
    </p>
  </div>
</div>
`;
